File: common/src/main/java/scooterdelta/challenge/bot/common/lookup/ShipType.java
package scooterdelta.challenge.bot.common.lookup;

import com.google.gson.annotations.SerializedName;

public enum ShipType {
    @SerializedName("Battleship")
    BATTLESHIP,
    @SerializedName("Carrier")
    CARRIER,
    @SerializedName("Cruiser")
    CRUISER,
    @SerializedName("Destroyer")
    DESTROYER,
    @SerializedName("Submarine")
    SUBMARINE
}

File: common/src/main/kotlin/scooterdelta/challenge/bot/common/lookup/ShipType.kt
package scooterdelta.challenge.bot.common.lookup

import com.fasterxml.jackson.annotation.JsonProperty

enum class ShipType {
    @JsonProperty("Battleship")
    BATTLESHIP,
    @JsonProperty("Carrier")
    CARRIER,
    @JsonProperty("Cruiser")
    CRUISER,
    @JsonProperty("Destroyer")
    DESTROYER,
    @JsonProperty("Submarine")
    SUBMARINE
}